lib/features.js

```diff
@@ -64,6 +64,40 @@ module.exports = config => {
       return isSampledUser(signinConfirmation.sample_rate, uid, 'signinConfirmation')
     },
 
+
+    /**
+     * Returns whether or not to use signin unblock feature on a request.
+     *
+     * @param account
+     * @param config
+     * @param request
+     * @returns {boolean}
+     */
+    isSigninUnblockEnabledForUser(uid, email, request) {
+      if (! signinUnblock.enabled) {
+        return false
+      }
+
+      // Or if the email address matching one of these regexes.
+      if (signinUnblock.allowedEmailAddresses.test(email)) {
+        return true
+      }
+
+      // While we're testing this feature, there may be some funky
+      // edge-cases in device login flows that haven't been fully tested.
+      // Temporarily avoid them for regular users by checking the `context` flag,
+      const context = request.payload &&
+        request.payload.metricsContext &&
+        request.payload.metricsContext.context
+
+      if (signinUnblock.supportedClients.indexOf(context) === -1) {
```

Comment:
If the list is empty, should it be enabled for everyone? The second question is how can we make it so it forces users w/o a context through the flow too? That'd be fore users that just arrive at `accounts.firefox.com` and sign-in but are blocked.